pow-migration/src/lib.rs

```diff
@@ -302,7 +305,24 @@ pub async fn migrate(
             )
             .await?;
 
-            log::info!("PoS Genesis generation is completed");
+            // Sort vectors for a consistent hash digest
+            genesis_config.validators.sort();
+            genesis_config.stakers.sort();
+            genesis_config.basic_accounts.sort();
+            genesis_config.vesting_accounts.sort();
+            genesis_config.htlc_accounts.sort();
+
+            let mut hasher = Blake2bHasher::new();
+            genesis_config
+                .serialize_to_writer(&mut hasher)
+                .unwrap_or_else(|error| {
+                    exit_with_error(error, "Failed to serialize genesis config")
+                });
+            genesis_config_hash = hasher.finish();
+            log::info!(
+                genesis_config_hash = genesis_config_hash.to_hex(),
```

Comment:
`Blake2bHash` implements `Display`
```suggestion
                %genesis_config_hash,
```